test/log-iterator.spec.js

```diff
@@ -0,0 +1,243 @@
+'use strict'
+
+const assert = require('assert')
+const rmrf = require('rimraf')
+const Log = require('../src/log')
+const AccessController = Log.AccessController
+const IdentityProvider = require('orbit-db-identity-provider')
+
+// Test utils
+const {
+  config,
+  testAPIs,
+  startIpfs,
+  stopIpfs
+} = require('./utils')
+
+let ipfs, testIdentity
+
+Object.keys(testAPIs).forEach((IPFS) => {
+  describe('Log - Iterator (' + IPFS + ')', function () {
+    this.timeout(config.timeout)
+
+    const testACL = new AccessController()
+    const { identityKeysPath, signingKeysPath } = config
+    const ipfsConfig = Object.assign({}, config.defaultIpfsConfig, {
+      repo: config.defaultIpfsConfig.repo + '-log-join' + new Date().getTime()
+    })
+
+    before(async () => {
+      rmrf.sync(ipfsConfig.repo)
+      testIdentity = await IdentityProvider.createIdentity({ id: 'userA', identityKeysPath, signingKeysPath })
+      ipfs = await startIpfs(IPFS, ipfsConfig)
+    })
+
+    after(async () => {
+      await stopIpfs(ipfs)
+      rmrf.sync(ipfsConfig.repo)
+    })
+
+    describe('Basic iterator functionality', () => {
+      let log1
+
+      beforeEach(async () => {
+        log1 = new Log(ipfs, testIdentity, { access: testACL, logId: 'X' })
+
+        for (let i = 0; i <= 100; i++) {
+          await log1.append('entry' + i)
+        }
+      })
```

Comment:
Would be good to test with forking/joining log also, eg. with this https://github.com/orbitdb/ipfs-log/blob/master/test/utils/log-creator.js#L6. It may reveal some new design considerations and edge cases.

Reply:
Added tests for forked/joins based on the logCreator work. Have a look and resolve this convo if you like it :)

Reply:
Looks good now 👍 Thanks!